polkadot/node/network/statement-distribution/src/v2/mod.rs

```diff
@@ -754,22 +794,7 @@ fn find_active_validator_state(
 	let our_group = groups.by_validator_index(validator_index)?;
 
 	let core_index = group_rotation_info.core_for_group(our_group, availability_cores.len());
-	let paras_assigned_to_core = if let Some(claim_queue) = maybe_claim_queue {
-		claim_queue.iter_claims_for_core(&core_index).copied().collect()
-	} else {
-		availability_cores
-			.get(core_index.0 as usize)
-			.and_then(|core_state| match core_state {
-				CoreState::Scheduled(scheduled_core) => Some(scheduled_core.para_id),
-				CoreState::Occupied(occupied_core) if max_candidate_depth >= 1 => occupied_core
-					.next_up_on_available
-					.as_ref()
-					.map(|scheduled_core| scheduled_core.para_id),
-				CoreState::Free | CoreState::Occupied(_) => None,
-			})
-			.into_iter()
-			.collect()
-	};
+	let paras_assigned_to_core = claim_queue.iter_claims_for_core(&core_index).copied().collect();
```

Comment:
shouldn't we preserve the functionality for synchronous backing (max_candidate_depth 0)? In this case we shouldn't take into account cores that are occupied

Reply:
We should remove all these fallbacks to simplify the code.